File: MyDent.Services/Database/DentalService.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MyDent.Services.Database
{
    public class DentalService
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(150)]
        public string Name { get; set; } = string.Empty;

        [MaxLength(1000)]
        public string Description { get; set; } = string.Empty;

        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal Price { get; set; }

        [Required]
        public int DurationMinutes { get; set; }

        public bool IsActive { get; set; } = true;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public int ServiceCategoryId { get; set; }

        [ForeignKey("ServiceCategoryId")]
        public ServiceCategory ServiceCategory { get; set; } = null!;

        public int? ImageAssetId { get; set; }

        [ForeignKey("ImageAssetId")]
        public Asset? Image { get; set; }

        public ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();
    }
}

File: UI/mydent_mobile/lib/models/dental_service.dart
import 'package:json_annotation/json_annotation.dart';

part 'dental_service.g.dart';

@JsonSerializable()
class DentalService {
  final int? id;
  final String? name;
  final String? description;
  final double? price;
  final int? durationMinutes;
  final bool? isActive;
  final DateTime? createdAt;
  final int? serviceCategoryId;
  final String? serviceCategoryName;
  final int? imageAssetId;

  DentalService({
    this.id,
    this.name,
    this.description,
    this.price,
    this.durationMinutes,
    this.isActive,
    this.createdAt,
    this.serviceCategoryId,
    this.serviceCategoryName,
    this.imageAssetId,
  });

  factory DentalService.fromJson(Map<String, dynamic> json) =>
      _$DentalServiceFromJson(json);

  Map<String, dynamic> toJson() => _$DentalServiceToJson(this);
}
